docs/TypeConverters/index.md

```diff
@@ -0,0 +1,148 @@
+
+```csharp
+class SomeClass
+{
+	string m_ID;
+
+	int GetID()
+	{
+		auto converter = CF_TypeConverter.Get(string);
+
+		converter.Read(this, "m_ID");
+
+		return converter.GetInt();
+	}
+
+	void SetID(int id)
+	{
+		auto converter = CF_TypeConverter.Get(string);
+
+		converter.SetInt(id);
+
+		converter.Write(this, "m_ID");
+	}
+}
+```
+
+## Custom Type Converters
+
+The following code example shows how to create and register an instance of a type converter. Do not create member variables as the instance of the type converter is a singleton and will be the same instance everytime it is used. Only one type converter can be registered per distinct type. 
+
+```csharp
+[CF_RegisterTypeConverter(CF_TypeConverterPlayerStatBase)]
+class CF_TypeConverterPlayerStatBase : CF_TypeConverterClass
+{
+    // Insert code here.
+}
+```
+
+A common usage of type converters is to convert an object to a primitive type. The following code shows how `PlayerStatBase` can be setup to convert from floats and string primitive types. 
+
+```csharp
+[CF_RegisterTypeConverter(CF_TypeConverterPlayerStatBase)]
+class CF_TypeConverterPlayerStatBase : CF_TypeConverterClass
+{
+	override void SetInt(int value)
+	{
+		m_Value.SetByFloat(value);
+	}
+	
+	override int GetInt()
+	{
+		return m_Value.Get();
+	}
+
+	override void SetFloat(float value)
+	{
+		m_Value.SetByFloat(value);
+	}
+	
+	override float GetFloat()
+	{
+		return m_Value.Get();
+	}
+
+	override void SetString(string value)
+	{
+		m_Value.SetByFloat(value.ToFloat());
+	}
+
+	override string GetString()
+	{
+		return GetFloat().ToString();
+	}
+}
+```
+
+Another common usage is for class instances to be serialized and deserialized. The following code will only serialize and deserialize the float variable associated with `PlayerStatBase`.
+
+```csharp
+[CF_RegisterTypeConverter(CF_TypeConverterPlayerStatBase)]
+class CF_TypeConverterPlayerStatBase : CF_TypeConverterClass
+{
+    // Insert above example code here.
+
+	override bool Read(Serializer ctx)
+	{
+		float value;
+		if (!ctx.Read(value))
+			return false;
+
+		SetFloat(value);
+		
+		return true;
+	}
+
+	override bool Write(Serializer ctx)
+	{
+		return ctx.Write(GetFloat());
+	}
+}
+```
+
+To use the newly type converters you just have to get the instance and read back the variable.
+
+Unlike with normal type converters, 'Write' does not need to be called as the custom type converter written does not change the instance, just the variable within the instance.
+
+```csharp
+auto converter = CF_TypeConverter.Get(PlayerStatBase);
+converter.Read(GetGame().GetPlayer(), "m_StatDiet");
+
+Print(converter.GetFloat()); // will print out the current float variable associated with the player
+converter.SetString("0.5");
+Print(converter.GetFloat()); // will print out "0.5" as a float
+```
```

Comment:
For somebody that knows what he wants to do and just looks up the syntax maybe this works, but I have no idea in what situation I would want to create my own type converter and I don't know how to work with the m_Value, that is just in the code with no other explanation.

If I have my own data structs like SomeState that is essentially just an int stored as m_State, how do i convert between that and int/float/string?
```
class SomeState
{
	int m_State;
}
```
If I have to manually tell which member variable to access like you do here:
> converter.Read(GetGame().GetPlayer(), "m_StatDiet");

then what does type converter even do for me?

Reply:
> If I have to manually tell which member variable to access like you do here:

> converter.Read(GetGame().GetPlayer(), "m_StatDiet");

> then what does type converter even do for me?

It's the type of feature that wouldn't be used outside of #56 and #73

Reply:
For the record: @Jacob-Mango and I decided in a conversation to keep the docs surface level for now as its primary use is with a different feature that will explain the use-case better. We might want up update the docs for this later on to make the feature of type converters understandable on its own.